tests/kafkatest/tests/streams/streams_eos_upgrade_test.py

```diff
@@ -0,0 +1,255 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from ducktape.mark import matrix
+from ducktape.mark.resource import cluster
+from kafkatest.tests.kafka_test import KafkaTest
+from kafkatest.services.streams import StreamsEosTestDriverService, StreamsEosTestJobRunnerService, \
```

Comment:
This is a meta thought: in order to test the crash rolling bounce case, maybe we can leverage on the existing `StreamsTestBaseService#stop_node()` which allows either stop clean or not (SIG_KILL). I.e. instead of calling `processor_to_stop.stop()` we call `processor2.stop_nodes(not crash)`, and we add to the testing matrix a boolean whether the rolling bounce should clean stop or hard crash the app.

Reply:
Yes, absolutely. I wanted to add this, in case we decide to not inject errors. Done